test/foundry/EspressoTEEVerifier.t.sol

```diff
@@ -3,62 +3,121 @@ pragma solidity ^0.8.20;
+    bytes32 mrSigner = bytes32(0x0c8242bba090f54b10de0c2d1ca4b633b9c08b7178451c71d737c214b72fc836);
 
-        string memory tcbInfoPath = "/test/foundry/configs/tcbinfo.json";
-        string memory qeIdPath = "/test/foundry/configs/tee_identity.json";
-        qeIdDaoUpsert(3, qeIdPath);
-        fmspcTcbDaoUpsert(tcbInfoPath);
+    function setUp() public {
+        vm.createSelectFork("https://rpc.ankr.com/eth_sepolia");
+        // Get the instance of the DCAP Attestation QuoteVerifier on the Arbitrum Sepolia Rollup
+        vm.startPrank(adminTEE);
+        espressoTEEVerifier = new EspressoTEEVerifier(
+            mrEnclave,
+            mrSigner,
+            // Address of the deployed V3Verifier contract
+            address(0x6E64769A13617f528a2135692484B681Ee1a7169)
+        );
+        vm.stopPrank();
     }
 
-    /*
-      Test that the verify function returns sucess for a valid quote
-    */
     function testVerifyQuoteValid() public {
         vm.startPrank(adminTEE);
 
         string memory quotePath = "/test/foundry/configs/attestation.bin";
         string memory inputFile = string.concat(vm.projectRoot(), quotePath);
         bytes memory sampleQuote = vm.readFileBinary(inputFile);
-        bool success = espressoTEEVerifier.verify(sampleQuote);
+        bool success = espressoTEEVerifier.verify(sampleQuote, reportDataHash);
         assertEq(success, true);
         vm.stopPrank();
     }
 
-    /*
-      Test that the verify function returns false for an invalid quote
-    */
     function testVerifyQuoteInValid() public {
         string memory quotePath = "/test/foundry/configs/incorrect_attestation_quote.bin";
         string memory inputFile = string.concat(vm.projectRoot(), quotePath);
         bytes memory invalidQuote = vm.readFileBinary(inputFile);
-        bool success = espressoTEEVerifier.verify(invalidQuote);
+        bool success = espressoTEEVerifier.verify(invalidQuote, reportDataHash);
+        assertEq(success, false);
+    }
+
+    /**
+        Test incorrect report data hash
+    */
+    function testIncorrectReportDataHash() public {
+        vm.startPrank(adminTEE);
+
+        string memory quotePath = "/test/foundry/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        bool success = espressoTEEVerifier.verify(sampleQuote, bytes32(0));
+        assertEq(success, false);
+    }
+
+    function testIncorrectMrEnclave() public {
+        vm.startPrank(adminTEE);
+
+        string memory quotePath = "/test/foundry/configs/attestation.bin";
+        string memory inputFile = string.concat(vm.projectRoot(), quotePath);
+        bytes memory sampleQuote = vm.readFileBinary(inputFile);
+        espressoTEEVerifier = new EspressoTEEVerifier(
+            bytes32(0x51dfe95acffa8a4075b716257c836895af9202a5fd56c8c2208dacb79c659ff1),
```

Comment:
Define a variable like `badMrEnclave` here? And similarly for the other tests. Makes it a bit easier to read especially since the value is the same except for the last digit.

To be more certain that we are always reverting for the right reasons I think it would be a bit better to have custom Error types for each reason we fail verification and check in the tests if we are reverting for the right reasons. Suggestion for later cleanup work.

Reply:
I will clean this up, might as well do it here!

Reply:
https://github.com/EspressoSystems/nitro-contracts/pull/28/commits/64c3051160e93d3ea330f72525d5aff308e28c16